lib/ios-log.js

```diff
@@ -0,0 +1,235 @@
+
+import spawn from 'win-spawn';
+import through from 'through';
+import path from 'path';
+import fs from 'fs';
+import _ from 'lodash';
+import which from 'which';
+import glob from 'glob';
+import { getLogger } from 'appium-logger';
+import mkdirp from 'mkdirp';
+import touch from 'touch';
+import xcode from 'xcode'; 
+
+// Date-Utils: Polyfills for the Date object
+
+let START_TIMEOUT = 10000;
+let logger = getLogger('iOS-log');
+
+class IosLog { 
+  constructor(opts) {
+    this.udid = opts.udid;
+    this.simUdid = opts.simUdid;
+    this.showLogs = opts.showLogs;
+    this.proc = null;
+    this.onIosLogStart = null;
+    this.iosLogStarted = false;
+    this.iosLogStartTime = null;
+    this.calledBack = false;
+    this.loggingModeOn = true;
+    this.logs = [];
+    this.logRow = '';
+    this.logsSinceLastRequest = [];
+    this.startTimer = null;
+  }
+  async startCapture () {
+    if (this.udid) { 
+      this.loggingModeOn = false;
+      let spawnEnv = _.clone(process.env);
+      logger.debug('Attempting iOS device log capture via libimobiledevice idevicesyslog');
+      let err = await which('idevicesyslog');
+      if (!err) { 
+        this.proc = spawn('idevicesyslog', {env: spawnEnv});
+      } else {
+        logger.warn('Could not capture device log using libimobiledevice idevicesyslog.' +
+                    ' Libimobiledevice probably is not installed');
+        logger.debug('Attempting iOS device log capture via deviceconsole');
+        let limDir = path.resolve(__dirname, '../../../build/deviceconsole');  //??? 
+        spawnEnv.PATH = `${process.env.PATH}:${limDir}`;
+        spawnEnv.DYLD_LIBRARY_PATH = limDir + ':' + process.env.DYLD_LIBRARY_PATH;
+        this.proc = spawn('deviceconsole', ['-u', this.udid], {env: spawnEnv});
+      }
+      this.finishStartingLogCapture();
+    } else {
+      let xcodeVersion = await xcode.getVersion();
+      let ver = parseInt(xcodeVersion.split('.')[0], 10);
+      if (ver >= 5) {
+        let logsPath;
+        if (ver >= 6) {
+          logger.debug('Starting iOS 8.* simulator log capture');
+          if (_.isUndefined(this.simUdid)) {
+            logger.errorAndThrow('ios8 log capture requires a sim udid'); 
+          }
+          logsPath = path.resolve(process.env.HOME, 'Library', 'Logs', 'CoreSimulator', this.simUdid); 
+        } else {
+          logger.debug('Starting iOS7.* simulator log capture');
+          logsPath = path.resolve(process.env.HOME, 'Library', 'Logs', 'iOS Simulator', '7.*');
+        }
+
+        let touchLog = async function () {
```

Comment:
part of the reason we created all of these intermediate functions (touchLog, getLogFiles, etc...), was because in ES5 callback land, it made things more readable. now with async/await, it actually makes it harder to understand. personally I think we might want to do away with the intermediate functions and first see how it looks just with all the code in one long block.

if it ends up being too long we can always break it up later, but then it will be according to rationale that makes sense for our new situation.